### PR TITLE
fix: high CPU usage on opening tarp #9390

### LINKED ISSUE #9390: fix(jans-tarp): high CPU usage on opening tarp
image

### PR DESCRIPTION
**Pull request by duttarnab**
closes #9390

### CODE REVIEW

File: demos/jans-tarp/src/options/authFlowInputs.tsx [hunk 1]
```diff
@@ -181,6 +181,7 @@ export default function AuthFlowInputs({ isOpen, handleDialog, client }) {
           console.log("userDetails: " + JSON.stringify(userInfoResponse.data));
```
Comment (mjatin-dev) [inline demos/jans-tarp/src/options/authFlowInputs.tsx:181]: Please remove console.logs.

File: demos/jans-tarp/src/options/oidcClients.tsx [hunk 2]
```diff
@@ -157,9 +172,9 @@ export default function OIDCClients(data) {
                             </TableRow>
                         </TableHead>
                         <TableBody>
-                            {(data.data === undefined || data.data.length == 0) ?
+                            {(data === undefined || data.length == 0) ?
                                 <TableCell colSpan={6}><Alert severity="warning">No Records to show.</Alert></TableCell> :
-                                data.data.map((row) => (<Row key={row.clientId} row={row} />))
+                                data.map((row) => (<Row key={row.clientId} row={row} notifyOnDataChange={handleNotifyOnDataChange} />))
```
Comment (mjatin-dev) [inline demos/jans-tarp/src/options/oidcClients.tsx]: Please check data length first. For example - data?.length ? data.map... : <>Not row found</>

File: demos/jans-tarp/src/options/userDetails.tsx [hunk 3]
```diff
@@ -61,23 +62,23 @@ const UserDetails = (data) => {
             </div>
             <legend><span className="number">O</span> User Details:</legend>
             <hr />
-            {data.data.displayToken ?
+            {data.displayToken ?
```
Comment (mjatin-dev) [inline demos/jans-tarp/src/options/userDetails.tsx]: Please check displayToken is exist in data for not. For example - data?.displayToken.

File: demos/jans-tarp/src/options/oidcClients.tsx [hunk 4]
```diff
@@ -88,11 +88,16 @@ function Row(props: { row: ReturnType<typeof createData> }) {
                 chrome.storage.local.set({ oidcClients: clientArr.filter(obj => obj.clientId !== row.clientId) });
             }
         });
+        handleNotifyOnDataChange();
     }
 
+    const handleNotifyOnDataChange = () => {
```
Comment (mjatin-dev) [inline demos/jans-tarp/src/options/oidcClients.tsx]: This function is unnecessary, you can directly use **notifyOnDataChange** function directly.

File: demos/jans-tarp/src/options/oidcClients.tsx [hunk 5]
```diff
@@ -123,17 +133,22 @@ function Row(props: { row: ReturnType<typeof createData> }) {
...
     const handleDialog = (isOpen) => {
         setModelOpen(isOpen);
+        notifyOnDataChange();
     };
 
     const handleDrawer = (isOpen) => {
         setDrawerOpen(isOpen);
     };
 
+    const handleNotifyOnDataChange = () => {
```
Comment (mjatin-dev) [inline demos/jans-tarp/src/options/oidcClients.tsx]: This function is unnecessary, you can directly use notifyOnDataChange function.

File: demos/jans-tarp/src/options/oidcClients.tsx (hunk 2 above)
Comment (duttarnab) [reply demos/jans-tarp/src/options/oidcClients.tsx:177]: Please let it be as it is , it is for displaying user-info on console.

File: demos/jans-tarp/src/options/authFlowInputs.tsx (hunk 1 above)
Comment (duttarnab) [reply demos/jans-tarp/src/options/authFlowInputs.tsx:181]: Please let it be as it is , it is for displaying user-info on console.

File: demos/jans-tarp/src/options/oidcClients.tsx (hunk 2 above)
Comment (duttarnab) [reply demos/jans-tarp/src/options/oidcClients.tsx]: done

File: demos/jans-tarp/src/options/userDetails.tsx (hunk 3 above)
Comment (duttarnab) [reply demos/jans-tarp/src/options/userDetails.tsx]: done

File: demos/jans-tarp/src/options/oidcClients.tsx (hunk 5 above)
Comment (duttarnab) [reply demos/jans-tarp/src/options/oidcClients.tsx]: done

**Review by mjatin-dev — APPROVED** [review]

**Review by moabu — APPROVED** [review]

**Pull request merged**